docs/guide/topics/error_handling.rst

```diff
@@ -0,0 +1,361 @@
+
+- Using the :meth:`Cog.has_error_handler <discord.ext.commands.Cog.has_error_handler>`
+  and :meth:`Command.has_error_handler <discord.ext.commands.Command.has_error_handler>`
+  methods in your global error handler.
+
+**Example**
+
+.. code-block:: python3
+
+    @bot.listen()
+    async def on_command_error(ctx, error):
+        """ Global error handler that ignores local error handlers. """
+
+        if ctx.cog and ctx.cog.has_error_handler():
+            # If the cog has an error handler, we return,
+            # ignoring the error.
+            return
+        
+        if ctx.command and ctx.command.has_error_handler():
+            # We do the same for the command.
+            return
+        
+        # Simpler way to handle CommandInvokeError using getattr
+        error = getattr(error, 'original', error)
+
+        # Then we continue with our error handling as normal...
+        print(f"Ignoring unhandled exception in command {ctx.command!r}")
+        traceback.print_exception(type(error), error, error.__traceback__)
+
+- Attaching flags to the :class:`Context <~.ext.commands.Context>` to denote the error has been handled.
+
+**Example**
+
+*1. In the local error handler, we attach an attribute to our Context:*
+
+.. code-block:: python3
+
+    @ban.error
+    async def ban_error(ctx, error):
+        """ Handles only the MissingRequiredArgument error. """
+
+        if isinstance(error, commands.MissingRequiredArgument):
+            # We set an attribute to the context that will then
+            # be detected in the global error handler.
+            ctx.ignore_error = True
+
+            # Then we do the rest
+            await ctx.send(f'Please mention the person to ban using `{ctx.clean_prefix}ban @member`')
+
+*2. Then, in our error handler, we use ``getattr()`` to check for the custom attribute we just set:*
+
+..
+    Note, I couldn't get to reference getattr() to the python docs... >_<
+
+.. code-block:: python3
+
+    @bot.listen('on_command_error')
+    async def global_error_handler(ctx, error):
+        """ Global error handler that handles errors that aren't maked as ignored"""
+
+        # Check if `ctx.ignore_error` is True
+        if getattr(ctx, 'ignore_error', False) is True:
+            return  # we ignore the error
+
+        # Simpler way to handle CommandInvokeError using getattr
+        error = getattr(error, 'original', error)
+
+        # Then we continue with our error handling as normal...
+        print(f"Ignoring unhandled exception in command {ctx.command!r}")
+        traceback.print_exception(type(error), error, error.__traceback__)
+
+In :ref:`Events <discord-api-events>`
+-------------------------------------
+
+If an error ocurrs in an event, it does not go to any command error handler, as it is not a command. These errors are instead
+propagated to :meth:`~Client.on_error`, and can be retrieved with a standard call to :func:`sys.exc_info`. 
```

Comment:
Mentioning on_error's event-only behavior is important enough to drag to the top. Hiding it under the example is probably not a great move